### PR TITLE
Enabled torch.unique for bool tensors

### PR DESCRIPTION
**Pull request by izdeby**
Enabled torch.unique for bool tensors. 
Tested via unit tests

[issue](https://github.com/pytorch/pytorch/issues/27691)

### CODE REVIEW

**Comment by VitalyFedyunin** [comment]
Still multiple flake8 issues

**Comment by izdeby** [comment]
@VitalyFedyunin, all green

File: test/test_torch.py [hunk 1]
```diff
@@ -11360,8 +11360,14 @@ def run_test(device, dtype):
                                                   [2., 1.]]],
                                                 dtype=dtype,
                                                 device=device)
+            expected_unique_dim1_bool = torch.tensor([[[False, True], [True, True]],
```
Comment (ezyang) [inline test/test_torch.py]: @mruberry the code here is kind of awkward and it would be good if dtype genericity had a good way to handle this.

**Review by ezyang — APPROVED** [review]

File: test/test_torch.py (hunk 1 above)
Comment (mruberry) [reply test/test_torch.py]: Handle what, in particular? That the constructor for bool tensors is distinct?

Comment (ezyang) [reply test/test_torch.py]: Err, yes I guess? I'm not too sure what to do, I just know the code here looks bad.

**Pull request closed without merging**